queue_processors/autoreduction_processor/post_process_admin.py

```diff
@@ -206,12 +214,62 @@ def specify_instrument_directories(self,
         # Specify directories where autoreduction output will go
         return temporary_directory + instrument_output_directory
 
-    def reduction_started(self):
-        """Log and update AMQ message to reduction started"""
-        logger.debug("Calling: %s\n%s",
-                     ACTIVEMQ_SETTINGS.reduction_started,
-                     self.message.serialize(limit_reduction_script=True))
-        self.client.send(ACTIVEMQ_SETTINGS.reduction_started, self.message)
+    @staticmethod
+    def write_and_readability_checks(directory_list, read_write):
+        """
+        Check a list of N directories for write/readability
+        :param directory_list: (list) directory list
+        :param read_write: (str) Read=R, Write=W
+        :return Error (Exception or ValueError) when something goes wrong
+        """
+        read_write = read_write.upper()
+        read_write_map = {"R": "read", "W": "write"}
+        try:
+            read_write_map[read_write]
+        except KeyError:
+            raise KeyError("Invalid read or write input: %s read_write argument must be either"
+                           " 'R' or 'W'" % read_write)
+
+        for location in directory_list:
+            if not os.access(location, getattr(sys.modules[os.__name__], f"{read_write}_OK")):
+                if not os.access(location, os.F_OK):
+                    problem = "does not exist"
+                else:
+                    problem = "no %s access", read_write_map[read_write]
+                raise OSError("Couldn't %s %s  -  %s" % (read_write_map[read_write],
+                                                         location,
+                                                         problem))
+            return True
+
+    def path_access_validate(self, should_be_writable, should_be_readable):
+        """
+        Test for access to result paths and raise exception and attempt re-run later if problem
+        :param should_be_writable: (list)
+        :param should_be_readable: (list)
+        """
+        try:
+            # Try to make directories which should exist
+            for path in filter(lambda p: not os.path.isdir(p), should_be_writable):
```

Comment:
I think this would be easier to read as a list comp i.e.
```python
for path in [p for p in should_be_writable if not os.path.isdir(p)]:
    os.makedirs(path)
```
Though this is completely subjective.

Reply:
This is not code i have changed. It is the result of merging Master into 625. I will consider changing this later.